### ISSUE TITLE
Support for GraalVM native images

### ISSUE THREAD

**Issue by ilopmar** [liquibase/liquibase#1552]
Is there any plan to support [GraalVM](https://www.graalvm.org/) native images out of the box in Liquibase? At this moment different frameworks like Micronaut and Quarkus already support it but including specific Liquibase configuration into the frameworks or doing some tricky things.

It would be good if there were support directly in the library so anyone that uses it can benefit.

In Micronaut 2.2.0 that we released yesterday we upgraded Liquibase to 4.2.0 and we've added support for GraalVM. We have included [these two GraalVM configuration files](https://github.com/micronaut-projects/micronaut-liquibase/tree/master/liquibase/src/main/resources/META-INF/native-image/io.micronaut.liquibase/micronaut-liquibase) declaring a lot of classes for reflection usage. 
I don't know the internals of Liquibase so maybe we have included more classes than necessary or even worse, we are missing something.

If you are interested I can contribute a PR to include those files directly into the library so all users can benefit from it :-)

**Comment by molivasdat** [comment]
Hi @ilopmar Thanks for this enhancement request.  I'll check with the team on the plan for other integrations. If you have a PR that would resolve this then go ahead and submit it.  I'm not familiar with GraalVM so I will need to do some research.  This was not on the current radar as far as I know.

**Comment by ilopmar** [comment]
@molivasdat done in #1582

**Comment by nvoxland** [comment]
Because Liquibase relies so heavily on reflection, the GraalVM config will be complex to maintain in a generic/complete case. Until we better understand the level of interest, we'll just keep #2005 open for PRs, but will not bring that configuration into an official release. That PR includes the changes done in #1582 

I'll update the description with this information

**Comment by ilopmar** [comment]
Oh, I'm sorry to hear this but on the other hand I understand it. Let's hope the community is interested in getting official support for Liquibase and GraalVM and soon you can work on this. 
Thanks!

**Comment by nvoxland** [comment]
Yeah, would be great to support it but we also have a lot of other things we're needing to get through too. 

Feel free to help keep the branch up to date so it's ready when the demand is overwhelming :)

**Comment by sgammon** [comment]
@nvoxland this issue forces us to run liquibase manually rather than with our application server on boot. even if liquibase merely provides a starting point in configuration (with the user responsible for opening up their own classes), it would be really helpful / useful

**Comment by derkoe** [comment]
See also https://github.com/oracle/graalvm-reachability-metadata/pull/98/

**Comment by jmart1** [comment]
Does anyone know if FlyWay works with Liquidbase?  Asking here cause I imagine more people will have this problem and need a solution and might see this message.

**Comment by ctoabidmaqbool1** [comment]
@jmart1 Hi! I am a long term user of Flyway Db, it's over 3 years or more, many peoples are asking about GraalVM native image support for Flyway, but unfortunatly still Flyway db is not yet suported by native-iamge e.g. https://github.com/flyway/flyway/issues/2927.
I was looking alternative soutions for Graalvm native-image, but it looks like Liquibase is still not supported!!!
